patches/tModLoader/Terraria.ModLoader.Recipes/RecipeBuilder.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using Terraria;
+using Terraria.ModLoader;
+using Terraria.ModLoader.Exceptions;
+
+namespace Terraria.ModLoader.Recipes
+{
+	public class RecipeBuilder
+	{
+		private readonly ModRecipe _recipe;
+
+
+		/// <summary>Creates a new instance with no ingredients and no result.</summary>
+		/// <seealso cref="Requires(int,int)"/>
+		/// <seealso cref="Build"/>
+		public RecipeBuilder()
+		{
+		}
+
+		/// <summary>Creates a new instance with no ingredients with the given item and stack as a result.</summary>
+		/// <param name="mod">The mod who owns the recipe.</param>
+		/// <param name="item">The item type.</param>
+		/// <param name="stack">The stack.</param>
+		/// <seealso cref="Requires(int,int)"/>
+		/// <seealso cref="Build"/>
+		public RecipeBuilder(Mod mod, int item, int stack = 1)
+		{
+			_recipe = new ModRecipe(mod);
+			_recipe.SetResult(item, stack);
+		}
+
+		/// <summary>Creates a new instance with no ingredients with the given item and stack as a result.</summary>
+		/// <param name="mod">The mod who owns the recipe.</param>
+		/// <param name="itemName">Name of the item.</param>
+		/// <param name="stack">The stack.</param>
+		/// <seealso cref="Requires(int,int)"/>
+		/// <seealso cref="Build"/>
+		public RecipeBuilder(Mod mod, string itemName, int stack = 1)
+		{
+			_recipe = new ModRecipe(mod);
+			_recipe.SetResult(mod, itemName, stack);
+		}
+
+		/// <summary>Creates a new instance with no ingredients with the given item and stack as a result.</summary>
+		/// <param name="item"></param>
+		/// <param name="stack"></param>
+		/// <seealso cref="Requires(int,int)"/>
+		/// <seealso cref="Build"/>
+		public RecipeBuilder(ModItem item, int stack = 1)
```

Comment:
I would get rid of this, it's a bit dangerous to just 'inherit' the mod of the item. You will be adding a recipe 'as another mod' if the item comes from another mod.

Reply:
This is the equivalent of `ModRecipe.SetResult(ModItem, int)`, which is pretty standard in most mods adding ModItems with recipes. I don't think this should be removed, but maybe fixed ? Adding a mod parameter perhaps ?

Reply:
Ok if it's the same as that one, then it's fine